CodingStandards/ProgramEntryPoint.cs

```diff
@@ -6,32 +6,42 @@
 using System.Threading.Tasks;
 
 class Program
-{
-        //This is the entry point for the program.
-        static int Main(string[] args)
-{
-    return writeMessage(dtNow, Console.Out);
-}
-
-        static DateTime dtNow = DateTime.Now;
-
-public static int writeMessage(DateTime CurrentDate, TextWriter writer)
-{
-    var intRetVal = 0; const Int32 MESSAGE_COUNT = 5; for (int MyIndex = 0; MyIndex < MESSAGE_COUNT; MyIndex++)
-    /*********************/
-    /* greater than 2016 */
-    /*********************/
-    if (CurrentDate.Year > 2016) writer.WriteLine("Happy New Year!");
-    else
-                writer.WriteLine("Hello From " + CurrentDate.Year);
-
-
-    //--------------------return the result value----------------------------
-
-
-
-
-    return intRetVal;
+{
+    static DateTime DTNow = DateTime.Now;
+
+    // This is the entry point for the program.
+    /// <summary>
+    /// Prints messages to the console
+    /// </summary>
+    /// <param name="args"></param>
+    /// <returns></returns>
+    static int Main(string[] args)
+    {
+        return WriteMessage(DTNow, Console.Out);
+    }
+
+    /// <summary>
+    /// Writes a message. Content depends on the current date.
+    /// </summary>
+    /// <param name="CurrentDate"></param>
+    /// <param name="writer"></param>
+    /// <returns></returns>
+    public static int WriteMessage(DateTime CurrentDate, TextWriter writer)
+    {
+        var resultValue = 0;
+        const int messageCount = 5;

```

Comment:
Conventionally, const variables are placed at the top of a method with a space after them. This is not a coding standards violation but would be a good way to organize code.